### PR TITLE
refactor: Replace tableMetaCache and undoLogManager with SPI

### PR DESCRIPTION
**Pull request by ph3636**
### Ⅰ. Describe what this PR did
Replace with SPI

### CODE REVIEW

**Comment by zjinlei** [comment]
Thanks!
Coming soon 1.0, priority review after 1.0.

**Comment by zjinlei** [comment]
I hope you can join the seata development team, i8vx5rv is my DingTalk ID, please add friends.

**Comment by ph3636** [comment]
thanks, My pleasure

File: server/src/main/java/io/seata/server/lock/LockerFactory.java [hunk 1]
```diff
@@ -39,75 +26,25 @@
...
-    protected static LockManager lockManager;
+    private static volatile LockManager lockManager;
 
     /**
      * Get lock manager.
      *
      * @return the lock manager
      */
-    public static final LockManager getLockManager() {
+    public static LockManager getLockManager() {
```
Comment (zjinlei) [inline server/src/main/java/io/seata/server/lock/LockerFactory.java]: Locker spi can be separated from this pr.
Please take a look at Mission 53. If you are interested, you can claim this mission.
https://www.yuque.com/cgbcac/aoni5y/uw0n5d

Comment (ph3636) [reply server/src/main/java/io/seata/server/lock/LockerFactory.java]: All right, I claim it

Comment (ph3636) [reply server/src/main/java/io/seata/server/lock/LockerFactory.java]: Locker spi already separated from this pr.

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java [hunk 2]
```diff
@@ -288,7 +281,7 @@ public void undo(DataSourceProxy dataSourceProxy, String xid, long branchId) thr
                             Collections.reverse(sqlUndoLogs);
                         }
                         for (SQLUndoLog sqlUndoLog : sqlUndoLogs) {
-                            TableMeta tableMeta = TableMetaCacheFactory.getTableMetaCache(dataSourceProxy).getTableMeta(
+                            TableMeta tableMeta = TableMetaCacheFactory.getTableMetaCache(dataSourceProxy.getDbType()).getTableMeta(
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java]: dataSourceProxy.getDbType() return SqlParser.getDbType, but in other places is returned we define constants, whether there is the risk of inconsistent?

Comment (ph3636) [reply rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java]: Yes, there is a problem. It has been corrected

**Pull request merged**